libsql-server/src/replication/primary/logger.rs

```diff
@@ -266,13 +273,16 @@ unsafe impl WalHook for ReplicationLoggerHook {
             let ctx = Self::wal_extract_ctx(wal);
             let runtime = tokio::runtime::Handle::current();
             if let Some(replicator) = ctx.bottomless_replicator.as_mut() {
-                let mut replicator = replicator.lock().unwrap();
-                let _prev = replicator.new_generation();
-                if let Err(e) =
-                    runtime.block_on(async move { replicator.snapshot_main_db_file().await })
-                {
-                    tracing::error!("Failed to snapshot the main db file during checkpoint: {e}");
-                    return SQLITE_IOERR_WRITE;
+                if let Some(replicator) = replicator.lock().unwrap().as_mut() {
```

Comment:
We want to abort the operation if we find out that the replicator is not there anymore, by returning an error. Same in all cases where we check for the replicator, including in `on_frame`